zerver/lib/actions.py

```diff
@@ -3642,6 +3642,8 @@ def do_remove_realm_alias(alias):
     realm = alias.realm
     domain = alias.domain
     alias.delete()
+    if RealmAlias.objects.filter(realm=realm).count() == 0:
+        do_set_realm_restricted_to_domain(realm, False)
```

Comment:
`do_set_realm_restricted_to_domain` calls `send_event`, so we're good there, except we should check it wasn't already set to False before calling it.